setup_pip.py

```diff
@@ -0,0 +1,165 @@
+
+CMAKE_EXE = os.environ.get('CMAKE_EXE', shutil.which('cmake'))
+
+
+def distutils_dir_name(dname):
+    """Returns the name of a distutils build directory"""
+    f = "{dirname}.{platform}-{version[0]}.{version[1]}"
+    return os.path.join('build', f.format(dirname=dname,
+                                          platform=sysconfig.get_platform(),
+                                          version=sys.version_info))
+
+
+class CMakeExtension(Extension):
+    def __init__(self, name, sourcedir=''):
+        Extension.__init__(self, name, sources=[])
+        self.sourcedir = os.path.abspath(sourcedir)
+
+
+class CMakeBuildExt(build_ext):
+    def build_extension(self, ext):
+        if isinstance(ext, CMakeExtension):
+            output_dir = os.path.abspath(
+                os.path.dirname(self.get_ext_fullpath(ext.name)))
+
+            build_type = 'Debug' if self.debug else 'Release'
+            cmake_args = [CMAKE_EXE,
+                          ext.sourcedir,
+                          '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + output_dir,
+                          '-DCMAKE_BUILD_TYPE=' + build_type]
+            cmake_args.extend(
+                [x for x in
+                 os.environ.get('CMAKE_COMMON_VARIABLES', '').split(' ')
+                 if x])
+
+            env = os.environ.copy()
+            if not os.path.exists(self.build_temp):
+                os.makedirs(self.build_temp)
+            subprocess.check_call(cmake_args,
+                                  cwd=self.build_temp,
+                                  env=env)
+            subprocess.check_call(['make', '-j', ext.name],
+                                  cwd=self.build_temp,
+                                  env=env)
+            print()
+        else:
+            super().build_extension(ext)
+
+
+class InstallHeaders(install_headers):
+    """
+    The built-in install_headers installs all headers in the same directory
+    This overrides that to walk a directory tree and copy the tree of headers
+
+    """
+
+    def run(self):
+        headers = self.distribution.headers
+        if not headers:
+            return
+        self.mkpath(self.install_dir)
+        for header in headers:
+            for dirpath, dirnames, filenames in os.walk(header):
+                for file in filenames:
+                    if file.endswith('.h'):
+                        path = os.path.join(dirpath, file)
+                        install_target = os.path.join(self.install_dir,
+                                                      path.replace(header, ''))
+                        self.mkpath(os.path.dirname(install_target))
+                        (out, _) = self.copy_file(path, install_target)
+                        self.outfiles.append(out)
+
+
+# setup does not clean up the build directory, so do it manually
+shutil.rmtree('build', ignore_errors=True)
+
+install_requires = [
+    'pandas>=0.20,<0.21',
+    'numba>=0.40.0dev,<0.41',
```

Comment:
We don't want to grab dev builds anymore and I don't believe they're published on pip anyway. We can also use 0.41.0 without issue.
```suggestion
    'numba>=0.40.0,<0.42',
```